src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -1275,32 +1374,40 @@ bool ConditionMgr::isSourceTypeValid(Condition* cond)
 {
     if (cond->SourceType == CONDITION_SOURCE_TYPE_NONE || cond->SourceType >= CONDITION_SOURCE_TYPE_MAX)
     {
-        LOG_ERROR("sql.sql", "Invalid ConditionSourceType %u in `condition` table, ignoring.", uint32(cond->SourceType));
+        LOG_ERROR("sql.sql", "%s Invalid ConditionSourceType in `condition` table, ignoring.", cond->ToString().c_str());
         return false;
     }
 
     switch (cond->SourceType)
     {
     case CONDITION_SOURCE_TYPE_TERRAIN_SWAP:
+    {
+        LOG_ERROR("sql.sql", "CONDITION_SOURCE_TYPE_TERRAIN_SWAP: is only for master branch, skipped");
```

Comment:
?

Reply:
PR not completed yet i add all of this shit then i start cleanup xD